lfs/transfer_queue.go

```diff
@@ -354,23 +361,26 @@ func (q *TransferQueue) batchApiRoutine() {
 		}
 
 		objs, adapterName, err := api.Batch(config.Config, transfers, q.transferKind(), transferAdapterNames)
+		tracerx.Printf("_> resp %v %v %v", objs, adapterName, err)
 		if err != nil {
 			if errors.IsNotImplementedError(err) {
 				git.Config.SetLocal("", "lfs.batch", "false")
-
 				go q.legacyFallback(batch)
 				return
 			}
 
-			if q.canRetry(err) {
-				for _, t := range batch {
-					q.retry(t.(Transferable))
+			var errOnce sync.Once
+			for _, o := range batch {
```

Comment:
If eventually we end up grouping retried objects and "fresh" objects in the same batch, we should only fail objects that have exceeded their retry budget, and not all of the items in the batch at once. To keep the old behavior of reporting to the errorc channel the same, I wrapped it in a sync.Once, so it can only happen once.